### PR TITLE
Support xz compressed packages

### LINKED ISSUE #2187: Support `xz` compressed source packages
Example failure:

```
$ uv pip install 'python-apt @ https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/python-apt/2.7.6/python-apt_2.7.6.tar.xz'
error: Failed to download and build: python-apt @ https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/python-apt/2.7.6/python-apt_2.7.6.tar.xz
  Caused by: Failed to extract source distribution
  Caused by: Unsupported archive type: python-apt_2.7.6.tar.xz
```

This is probably low priority, I'm sure.

### PR DESCRIPTION
**Pull request by krishnan-chandra**
## Summary

Closes #2187.

The [xz backdoor](https://gist.github.com/thesamesam/223949d5a074ebc3dce9ee78baad9e27) is still fairly recent, but luckily the [Rust `xz2` crate bundles version 5.2.5 of the C `xz` package](https://github.com/alexcrichton/xz2-rs/tree/main/lzma-sys), which is before the backdoor was introduced.

It's worth noting that a security risk still exists if you have a compromised version of `xz` installed on your system, but that risk is not introduced by `uv` or the Rust packages in general.

## Test Plan

Tried installing the package mentioned in the linked issue: `python-apt @ https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/python-apt/2.7.6/python-apt_2.7.6.tar.xz`

(Note that this will only work on Ubuntu - I tried on a Mac and while the archive was extracted properly, the package did not install because of some missing files)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -226,7 +226,7 @@ ci = ["github"]
 installers = ["shell", "powershell"]
 # The archive format to use for windows builds (defaults .zip)
 windows-archive = ".zip"
-# The archive format to use for non-windows builds (defaults .tar.xz)
+# The archive format to use for non-windows builds (defaults .tar.gz)
```
Comment (charliermarsh) [inline Cargo.toml:229]: I think this actually _is_ the default, but we changed it for uv below.

**Review by charliermarsh — APPROVED** [review]
Thanks!

**Comment by charliermarsh** [comment]
Nice PR.

File: Cargo.toml (hunk 1 above)
Comment (krishnan-chandra) [reply Cargo.toml]: Gotcha, this was just really confusing because the comment doesn't match the implementation. Maybe best to clarify.

**Pull request merged**